### ISSUE TITLE
Failure packaging google-cloud-cpp

### ISSUE THREAD

**Issue by cardigliano** [googleapis/google-cloud-cpp#8955]
**Does this issue affect the google-cloud-cpp project?**
Yes, this is about packaging google-cloud-cpp

**What component of `google-cloud-cpp` is this related to?**
This is related to the google-cloud-cpp installation

**Describe the bug**
I am following the instructions reported in doc/packaging.md,
after installing all dependencies (on Ubuntu 22) compilation seems to complete successful, 
instead the installation command fails (cmake --build cmake-out --target install)

**To Reproduce**

cmake -H. -Bcmake-out -DBUILD_TESTING=OFF -DGOOGLE_CLOUD_CPP_ENABLE_EXAMPLES=OFF
(this completes successfully)

cmake --build cmake-out --target install
[  1%] Built target googleapis_download
external/googleapis/CMakeFiles/google_cloud_cpp_api_backend_protos.dir/build.make:82: *** target pattern contains no '%'.  Stop.
gmake[1]: *** [CMakeFiles/Makefile2:604: external/googleapis/CMakeFiles/google_cloud_cpp_api_backend_protos.dir/all] Error 2
gmake: *** [Makefile:136: all] Error 2

**Expected behavior**
I expect the installation to complete and have the libraries available in my system.

**Operating system:**
Ubuntu 22 LTS

**What compiler and version are you using?**
gcc version 11.2.0 (Ubuntu 11.2.0-19ubuntu1) 

**What version of `google-cloud-cpp` are you using?**
Latest code from github: 85d0855afbaa55580c429bec8c76ee9e5e4e532c

**Additional context**

**Comment by coryan** [comment]
Thank you for the detailed bug report and for the build file.  I am not able to reproduce the problem, my build completed without problems.  FWIW, I used the dockerfile in #8957 to install the dependencies of `google-cloud-cpp`.

I noticed a significant difference is the `build.make` file.  Your version reads:

```Makefile
external/googleapis/google/api/backend.grpc.pb.cc: external/googleapis/src/googleapis_download/google/api/backend.proto
external/googleapis/google/api/backend.grpc.pb.cc: _protobuf_PROTOC_EXECUTABLE-NOTFOUND
external/googleapis/google/api/backend.grpc.pb.cc: gRPC::grpc_cpp_plugin-NOTFOUND
	@$(CMAKE_COMMAND) -E cmake_echo_color --switch=$(COLOR) --blue --bold --progress-dir=/home/cardigliano/nProbe/projects/gcloud-pubsub/google-cloud-cpp/cmake-out/CMakeFiles --progress-num=$(CMAKE_PROGRESS_2) "Running gRPC C++ protocol buffer compiler on /home/cardigliano/nProbe/projects/gcloud-pubsub/google-cloud-cpp/cmake-out/external/googleapis/src/googleapis_download/google/api/backend.proto"
	cd /home/cardigliano/nProbe/projects/gcloud-pubsub/google-cloud-cpp/cmake-out/external/googleapis && _protobuf_PROTOC_EXECUTABLE-NOTFOUND --plugin=protoc-gen-grpc=gRPC::grpc_cpp_plugin-NOTFOUND --grpc_out=/home/cardigliano/nProbe/projects/gcloud-pubsub/google-cloud-cpp/cmake-out/external/googleapis --cpp_out=/home/cardigliano/nProbe/projects/gcloud-pubsub/google-cloud-cpp/cmake-out/external/googleapis --proto_path /home/cardigliano/nProbe/projects/gcloud-pubsub/google-cloud-cpp/cmake-out/external/googleapis/src/googleapis_download --proto_path /usr/include /home/cardigliano/nProbe/projects/gcloud-pubsub/google-cloud-cpp/cmake-out/external/googleapis/src/googleapis_download/google/api/backend.proto
```

Mine reads:

```Makefile
external/googleapis/google/api/backend.grpc.pb.cc: external/googleapis/src/googleapis_download/google/api/backend.proto
external/googleapis/google/api/backend.grpc.pb.cc: /usr/local/bin/protoc-3.20.1.0
external/googleapis/google/api/backend.grpc.pb.cc: /usr/local/bin/grpc_cpp_plugin
        @$(CMAKE_COMMAND) -E cmake_echo_color --switch=$(COLOR) --blue --bold --progress-dir=/workspace/cmake-out/CMakeFiles --progress-num=$(CMAKE_PROGRESS_2) "Running gRPC C++ protocol buffer compiler on /workspace/cmake-out/external/googleapis/src/googleapis_download/google/api/backend.proto"
        cd /workspace/cmake-out/external/googleapis && /usr/local/bin/protoc-3.20.1.0 --plugin=protoc-gen-grpc=/usr/local/bin/grpc_cpp_plugin --grpc_out=/workspace/cmake-out/external/googleapis --cpp_out=/workspace/cmake-out/external/googleapis --proto_path /workspace/cmake-out/external/googleapis/src/googleapis
_download --proto_path /usr/local/include /workspace/cmake-out/external/googleapis/src/googleapis_download/google/api/backend.proto
```

That indicates (to me) that CMake was not able to find `protoc` in your environment, and could not find `grpc_cpp_plugin` either.

- Did you install Protobuf and gRPC?
- If so, are they in a location that your cmake configuration step (`cmake -H. -Bcmake-out -DBUILD_TESTING=OFF -DGOOGLE_CLOUD_CPP_ENABLE_EXAMPLES=OFF` above) can find them?
- Did you install them with the necessary CMake support files?

Hopefully that can help you fix the problems.  If not, please send us all the steps you used to install the dependencies, a Dockerfile would be great, but any form that is convenient for you can be made to work.

**Comment by coryan** [comment]
Looks more like a question to me, we can change it back to a bug once we reproduce the problem.

**Comment by cardigliano** [comment]
It seems that using the Dockerfile fixed the compilation, even if it is installing many dependencies from source code which is not my preferred solution, but it works :-) Thank you!

**Comment by coryan** [comment]
Using pre-built binaries is my (personal) preference too, unfortunately the version of Protobuf and gRPC shipped with Ubuntu (even Ubuntu:22.04) are just too old for our needs.
